tests/main/document-portal-activation/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Check that the document portal is activated when needed
+description: |
+    In order for xdg-document-portal to securely share files with a
+    confined applications, it must be started prior to setting up the
+    user mount namespace.  This is due to the daemon providing a FUSE
+    file system that needs to be bind mounted in the sandbox.
+
+    With that in mind, we don't want every snap invocation to try and
+    start the document portal.  Only in the following cases:
+
+        - a session bus is running
+        - the snap plugs the "desktop" interface
+
+    Furthermore, we don't want to print an error on systems where
+    xdg-document-portal is not available.
+
+systems: [ -ubuntu-core-* ]
+
+prepare: |
+    #shellcheck source=tests/lib/pkgdb.sh
+    . $TESTSLIB/pkgdb.sh
+    distro_install_package python3-dbus python3-gi
+    snap try "$TESTSLIB"/snaps/test-snapd-desktop
+    snap try "$TESTSLIB"/snaps/test-snapd-tools
+    rm -f /usr/share/dbus-1/services/fake-document-portal.service
+
+restore: |
+    [ -f dbus-launch.pid ] && kill "$(cat dbus-launch.pid)"
+    rm -f dbus-launch.pid
+    rm -f stderr.log doc-portal.log report-error.txt
+    rm -f /usr/share/dbus-1/services/fake-document-portal.service
+
+execute: |
+    echo "No output on stderr when running without a session bus"
+    user_data="$HOME/snap/test-snapd-desktop/current"
+    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
+    [ "$(wc -c < stderr.log)" -eq 0 ]
+
+    echo "Starting session bus"
+    eval "$(dbus-launch --sh-syntax)"
+    echo "$DBUS_SESSION_BUS_PID" > dbus-launch.pid
+
+    echo "No output on stderr when running with a session bus, when xdg-document-portal is not present."
+    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
+    [ "$(wc -c < stderr.log)" -eq 0 ]
+
+    echo "Make the fake document portal activatable"
+    cat << EOF > /usr/share/dbus-1/services/fake-document-portal.service
+    [D-BUS Service]
+    Name=org.freedesktop.portal.Documents
+    Exec=$(pwd)/fake-document-portal.py $(pwd)/doc-portal.log $(pwd)/report-error.txt
+    EOF
+    > report-error.txt
+
+    echo "No output on stderr when running with a session bus and xdg-document-portal is present".
+    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
+    MATCH "GetMountPoint called" < doc-portal.log
+    [ "$(wc -c < stderr.log)" -eq 0 ]
+
+    echo "Putting fake document portal into failure mode"
+    echo "error" > report-error.txt
+    > doc-portal.log
```

Comment:
This fails shellcheck. Can you update tests/unit/spread-shellcheck/must to include this task.yaml?

Reply:
Done.  I also enabled strict shellcheck for a few of the other tests I've added recently that were already compliant.

With that said, if the aim is to ratchet this down to nothing, it might be better to have a list of tests that are allowed to fail rather than a list of tests that must pass: that way new tests are automatically in the "must" category.